pkg/ratelimiter/rate_limiter.go

```diff
@@ -0,0 +1,67 @@
+package ratelimiter
+
+import (
+	"math"
+	"sync"
+	"time"
+)
+
+var (
+	DefaultBaseDelay = 2 * time.Minute
+	DefaultMaxDelay  = 10 * time.Minute
+	defaultMaxTries  = 3
+)
+
+type ItemExponentialFailureRateLimiterWithMaxTries struct {
+	// failuresLock mutex, which is necessary due to the fact that controller-runtime allows for concurrent reconciler invocations for a single controller
+	failuresLock sync.Mutex
+	failures     map[interface{}]int
+	maxTries     int
+
+	baseDelay time.Duration
+	maxDelay  time.Duration
+}
+
+func NewItemExponentialFailureRateLimiterWithMaxTries(baseDelay time.Duration, maxDelay time.Duration) RateLimiter {
+	return &ItemExponentialFailureRateLimiterWithMaxTries{
+		failures:  map[interface{}]int{},
+		baseDelay: baseDelay,
+		maxDelay:  maxDelay,
+		maxTries:  defaultMaxTries,
+	}
+}
```

Comment:
I think it would be better if there was an opportunity to configure `maxRetries` myself. If it is not configured then set `defaultMaxTries` value

Reply:
I think it's more of a generic thing. But let's get another opinion about this

Reply:
Not sure about the idea of configuring the `maxTries`. I think it's more of a generic thing. But let's get another opinion about this.